shaptools/hdb_connector/connectors/dbapi_connector.py

```diff
@@ -48,26 +48,25 @@ def connect(self, host, port=30015, **kwargs):
             )
         except dbapi.Error as err:
             raise base_connector.ConnectionError('connection failed: {}'.format(err))
-        self._logger.info('connected succesfully')
+        self._logger.info('connected successfully')
 
     def query(self, sql_statement):
         """
-        Query a sql statement and return response
+        Query a sql query result and return a result object
         """
         self._logger.info('executing sql query: %s' % sql_statement)
         try:
             with self._connection.cursor() as cursor:
                 cursor.execute(sql_statement)
-                result = cursor.fetchall()
+                result = base_connector.QueryResult().fetch_query(cursor)
```

Comment:
Right now this cannot work. You are setting to `result` the return of `fetch_query` and this method is not returning anything (no `return` keyword used). 

Change to:
```
result = base_connector.QueryResult.load_cursor(cursor)
```